File: first_primes/first_primes.rs
// 0.90s on rust 1.0.0-alpha
fn main() {
  let mut primes = vec![];
  let mut x: u32 = 2;
  let mut len: u32 = 0;
  while len < 10000 {
    let mut is_prime = true;
    for &prime in primes.iter() {
      if x % prime == 0 {
        is_prime = false;
        break;
      }
    }
    if is_prime {
      primes.push(x);
      len = len + 1;
    }
    x = x + 1;
  }
  for &v in primes.iter() {
    print!("{} ", v);
  } 
}

File: first_primes/first_primes.c
#include <stdio.h>

int main() {
  int primes[10000];
  int x = 2;
  int length;
  int i;
  int isPrime;
  for (length = 0; length < 10000; x++) {
    isPrime = 1;
    for (i = 0; i < length; i++) {
      if (x % primes[i] == 0) {
        isPrime = 0;
        break;
      }
    }
    if (isPrime) {
      primes[length] = x;
      length = length + 1;
    }
  }

  for (i = 0; i < length; i++) {
    printf("%d ", primes[i]);
  }
  return 0;
}
